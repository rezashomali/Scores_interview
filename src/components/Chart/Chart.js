import React from "react";
import {
  BarChart,
  Bar,
  XAxis,
  YAxis,
  Label,
  ResponsiveContainer,
  Tooltip,
} from "recharts";
import Title from "../Title/Title";

export default function Chart({ chartData, chartDataType }) {
  return (
    <React.Fragment>
      <Title>Score statistics</Title>
      <ResponsiveContainer>
        <BarChart
          data={chartData}
          margin={{
            top: 16,
            right: 16,
            bottom: 0,
            left: 24,
          }}
        >
          <Tooltip />
          <XAxis dataKey={chartDataType} />
          <YAxis>
            <Label angle={270} position="left" style={{ textAnchor: "middle" }}>
              Average score
            </Label>
          </YAxis>
          <Bar type="monotone" dataKey="score" fill="#556CD6" />
        </BarChart>
      </ResponsiveContainer>
    </React.Fragment>
  );
}
