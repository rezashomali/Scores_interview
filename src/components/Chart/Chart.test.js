import React from "react";
import renderer from "react-test-renderer";
import Chart from "./Chart";

function createNodeMock() {
  const doc = document.implementation.createHTMLDocument();
  return { parentElement: doc.body };
}

describe("Chart", () => {
  test("renders properly", () => {
    const data = [
      { country: "SE", score: 52.35 },
      { country: "US", score: 100 },
      { country: "EU", score: 100 },
    ];

    const tree = renderer
      .create(<Chart chartData={data} chartDataType="country" />, {
        createNodeMock,
      })
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
