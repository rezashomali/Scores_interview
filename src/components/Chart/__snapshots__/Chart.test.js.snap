// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart renders properly 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
  >
    Score statistics
  </h2>,
  <div
    className="recharts-responsive-container"
    style={
      Object {
        "height": "100%",
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "display": "none",
          "height": 0,
          "position": "absolute",
          "visibility": "hidden",
          "width": 0,
        }
      }
    />
  </div>,
]
`;
